src/java.base/share/classes/java/time/ZoneOffset.java

```diff
@@ -424,16 +428,24 @@ public static ZoneOffset ofTotalSeconds(int totalSeconds) {
             throw new DateTimeException("Zone offset not in valid range: -18:00 to +18:00");
         }
         if (totalSeconds % (15 * SECONDS_PER_MINUTE) == 0) {
-            return SECONDS_CACHE.computeIfAbsent(totalSeconds, totalSecs -> {
-                ZoneOffset result = new ZoneOffset(totalSecs);
-                ID_CACHE.putIfAbsent(result.getId(), result);
-                return result;
-            });
+            int slot = cacheSlot(totalSeconds);
+            ZoneOffset cached = SECONDS_CACHE.get(slot);
```

Comment:
I miss `AtomicReferenceArray::computeIfNull` that atomically will compute an element if the value at a certain index is `null`.

Reply:
@minborg You could CAS in a CompletableFuture—if you succeed you can complete it with the computation (bonus points since the computation can be done async) and if you read the value and get a CompletableFuture you can decide if you want to block on it, and for how long?